### PR TITLE
Connectors for data sources

### LINKED ISSUE #2086: Phase 4: Make Connectors to talk to data sources
### Phase 4 in the epic

Make Connectors that use this interface to talk to a WS or Web worker based backend for our UI:
```json
{ subscribe, unsubscribe, send, connect, disconnect }
```

The connectors should
* abstract the communication method so the UI doesn't have to know details after instantiation
* be performant
* allow a mock connector, for testing, to be easy to construct

### PR DESCRIPTION
**Pull request by brianosaurus**
Closes #2086 

Adding websocket, web worker connectors, and seo connectors

### CODE REVIEW

**Review by bthaile — APPROVED** [review]
lgtm

File: packages/augur-types/types/ethereum/events.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+export enum ContractEvents {
```
Comment (justinbarry) [inline packages/augur-types/types/ethereum/events.ts]: Can we move these to augur-artifacts?

Comment (brianosaurus) [reply packages/augur-types/types/ethereum/events.ts]: done.

**Review by justinbarry — APPROVED** [review]
LGTM

**Pull request merged**